### PR TITLE
feat: replace filter alt_allele_in_normal

### PR DESCRIPTION
**Pull request by mathiasbio**
## Description

This issue describes the problem with this filter: https://github.com/Clinical-Genomics/BALSAMIC/issues/1254
In short it is a very strict filter by TNscope that removes somatic variants sometimes even if they only have 1 read supporting it in the normal sample. The goal here is to remove it, and replace it with a less strict filter.

#### Added

- high_normal_tumor_af_frac filter in bcftools for TNscope WGS T+N analysis, and UMI T+N analysis which allows for 30% of tumor in the normal.

#### Changed

- [Description]

#### Fixed

- [Description]

#### Removed

- alt_allele_in_normal set by TNscope for WGS T+N analysis, and UMI T+N analysis

## Documentation

- [ ] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

## Tests

Detailed test-results and evaluations here: https://docs.google.com/spreadsheets/d/1_lNTCywOtEikJHXVhndfFzD1ieN21WSmQRDZVxVbAcY/edit#gid=0

Among other things:
- tumor_af vs normal_af plots comparing **alt_allele_in_normal** to the replacement filter **high_normal_tumor_af_frac** showing much more reasonable results for the new filter.
- sensitivity and precision analysis comparing these filters in TWISTpancancer reference with the following results:

From steadybedbug: **In the end filtering type 6 is used as it is more readable than type 5.**

| Filtering type | Step1                        | Step2                                                                                     | Threshold | True-pos-baseline | True-pos-call | False-pos | False-neg | Precision | Sensitivity | F-measure |
|----------------|------------------------------|-------------------------------------------------------------------------------------------|-----------|-------------------|---------------|-----------|-----------|-----------|-------------|-----------|
|              1 | alt_allele_in_normal removed | --e '(FORMAT/AF[1] / FORMAT/AF[0]) >= 0.3' --soft-filter high_normal_frac --mode +        | None      |                74 |            75 |       350 |        11 |    0.1765 |      0.8706 |    0.2935 |
|              2 | alt_allele_in_normal removed | -e sum(FORMAT/AF[0])<sum(FORMAT/AF[1])*4 -s high_normal_af_ratio -m +                     | None      |                84 |            85 |       359 |         1 |    0.1914 |      0.9882 |    0.3207 |
|              3 | alt_allele_in_normal removed |                                                                                           | None      |                85 |            86 |       640 |         0 |    0.1185 |           1 |    0.2118 |
|              4 | standard                     |                                                                                           | None      |                71 |            72 |       330 |        14 |    0.1791 |      0.8353 |     0.295 |
|              5 | alt_allele_in_normal removed | -e sum(FORMAT/AF[0])<sum(FORMAT/AF[1])*3.33 -s high_normal_af_ratio -m +                  | None      |                84 |            85 |       359 |         1 |    0.1914 |      0.9882 |    0.3207 |
|              6 | alt_allele_in_normal removed | filter -e sum(FORMAT/AF[1])/sum(FORMAT/AF[0])>0.3 --soft-filter high_normal_af_ratio -m + | None      |                84 |            85 |       359 |         1 |    0.1914 |      0.9882 |    0.3207 |

- Changes in number of variants between this PR and 14.0.0, in only quality-filtered VCF, and in the final clinical VCF:

| case        | workflow | # variants in vep/research VCF (v14.0.0) | # variants in vep/research VCF (this PR) | # variants in final VCF (v14.0.0) | # variants in final VCF (this PR) |
|-------------|----------|------------------------------------------|------------------------------------------|-----------------------------------|-----------------------------------|
| A  30% TINC | UMI T+N  |                                      302 |                                      331 |                               143 |                               198 |
| A  25% TINC | UMI T+N  |                                      302 |                                      329 |                               143 |                               196 |
| B 30% TINC  | WGS T+N  |                                     9098 |                                     9330 |                              7009 |                              7197 |

#### alt_allele_in_normal removal

Verify that filter alt_allele_in_normal is removed after quality-filtering in TNscope VCFs of tumor + normal analysis of WGS and UMI cases.
- [x] Verified

**Results in UMI TN-case: vcf/*.tnscope_umi.research.vcf.gz**

- [x] header contains bcftools command to remove alt_allele_in_normal
- [x] variant labelled with alt_allele_in_normal in raw VCF does not have filter in vcf/*.tnscope_umi.research.vcf.gz

**Results in WGS TN-case: vcf/*.tnscope.research.vcf.gz**

- [x] header contains bcftools command to remove alt_allele_in_normal
- [x] variant labelled with alt_allele_in_normal in raw VCF does not have filter in vcf/*.tnscope.research.vcf.gz

#### new high_normal_tumor_af_frac filter

Verify that new filter high_normal_tumor_af_frac is set correctly in example WGS T+N and UMI T+N cases.
- [x] Verified

**Results in UMI TN-case: vcf/*.tnscope_umi.research.vcf.gz**

- [x] header contains bcftools command to set high_normal_tumor_af_frac
- [x] variant with N-AF / T-AF smaller than 0.3 does not have filter set
- [x] variant with N-AF / T-AF greater than 0.3 has filter set

**Results in WGS TN-case: vcf/*.tnscope.research.vcf.gz**

- [x] header contains bcftools command to set high_normal_tumor_af_frac
- [x] variant with N-AF / T-AF smaller than 0.3 does not have filter set
- [x] variant with N-AF / T-AF greater than 0.3 has filter set

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [x] N/A
    - [ ] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [ ] N/A
    - [x] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [ ] N/A
    - [x] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

## Clinical Genomics Stockholm

<details>
<summary></summary>

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### User Changes

- [x] N/A
- [x] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [x] Affected users have been included in the development process and given a chance to provide feedback.
    - Email has been sent to customers, waiting for reply. (See mail-thread in SuSy: https://clinical-scilifelab.supportsystem.com/scp/tickets.php?id=67066)

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [x] N/A
    - [ ] Updated: [Link]
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

</details>

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [x] Provided a comprehensive description of the PR.
    - [x] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [x] Verified and updated documentation if necessary.
- **Tests**
    - [x] Described and tested the functionality addressed in the PR.
    - [x] Ensured integration of the new code with existing workflows.
    - [x] Confirmed that meaningful unit tests were added for the changes introduced.
    - [x] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [ ] Addressed and resolved all the feedback provided during the code review process.
    - [ ] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [x] Code implements the intended features or fixes the reported issue.
    - [x] Code follows the project's coding standards and style guide.
- **Documentation**
    - [x] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Tests**
    - [x] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [x] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

File: docs/balsamic_filters.rst [hunk 1]
```diff
@@ -195,6 +195,19 @@ The following filters are applied to the variants in TNscope raw VCF file (`SNV.
...
+::
+
+    bcftools annotate -x FILTER/alt_allele_in_normal
+
+
+*Relative tumor AF in normal*: Allows for maximum Tumor-In-Normal-Contamination of 30%.
+
+::
+
+    AF(tumor) / AF(normal) < 3.33
```
[the remark this answers is not in the document]
Comment (ivadym) [reply docs/balsamic_filters.rst:209]: and maybe we could say we're excluding based on that condition

**Review by ivadym — APPROVED** [review]
Looks good! 💯 

- The `bcftools annotate` command is used to exclude the `FILTER/alt_allele_in_normal annotation` from the VCF 
- Variants meeting the condition `AF[normal] / AF[tumor] > 0.3` will be excluded from further analysis as they are less likely to be relevant somatic mutations

File: BALSAMIC/constants/variant_filters.py [hunk 2]
```diff
@@ -50,6 +50,11 @@
         "field": "FORMAT",
     },
     "AF_min": {"tag_value": 0.05, "filter_name": "balsamic_low_af", "field": "FORMAT"},
+    "high_normal_tumor_af_frac": {
+        "tag_value": 3.33,
```
[the remark this answers is not in the document]
Comment (mathiasbio) [reply BALSAMIC/constants/variant_filters.py:54]: I know, and that was how I made it originally, but bcftools was acting odd and I found this issue in bcftools actually made by Hassan: https://github.com/samtools/bcftools/issues/1139 but I'll try again and see if I can make it more readable

[the remark this answers is not in the document]
Comment (ivadym) [reply BALSAMIC/constants/variant_filters.py:54]: Oh, then never mind. We can simply make it more straightforward in the documentation and leave that value as it is here

[the remark this answers is not in the document]
Comment (mathiasbio) [reply BALSAMIC/constants/variant_filters.py:54]: Ok it actually works, the important point was to use the sum() functions, then I could still use division and the fraction!

File: docs/balsamic_filters.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mathiasbio) [reply docs/balsamic_filters.rst]: right! everything is in include format usually

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,3 +1,9 @@
+[X.X.X]
+-------
+
```
[the remark this answers is not in the document]
Comment (mathiasbio) [reply CHANGELOG.rst:3]: Should I not add it? :D

[the remark this answers is not in the document]
Comment (ivadym) [reply CHANGELOG.rst:3]: Sorry, you missed to include the actual changes 😋

**Review by ivadym — APPROVED** [review]
👍
